common/SC_PaUtils.cpp

```diff
@@ -156,6 +161,11 @@ PaStreamParameters MakePaStreamParameters(int device, int channelCount, double s
     streamParams.channelCount = channelCount;
     streamParams.sampleFormat = fmt;
     streamParams.suggestedLatency = suggestedLatency;
+#ifdef __APPLE__
+    PaMacCore_SetupStreamInfo(&macInfo, paMacCorePro);
```

Comment:
do we really want this called on every invocation of this function?

Reply:
Maybe not, but it doesn't seem to hurt it. Should this be changed?

Reply:
just wondering. i'm a fan of keeping things as close to where they're used as possible. can you move `macInfo` inside this function as a `static` variable? then a reader doesn't need to jump up to the top of the file to find out what `macInfo` is.

Reply:
I think it needs to be the same `macInfo` instance, otherwise I was getting crashes. But I can investigate again...

Reply:
`static` would make it the same variable across calls and persistent for the life of the application. if it were a non-static local variable, it would be deallocated after the function call and you would probably get a segmentation fault later from the dangling pointer you stored in `streamParams`

see https://stackoverflow.com/questions/5033627/static-variable-inside-of-a-function-in-c for a quick explanation

Reply:
perfect! Thank you for that clarification, this is very helpful. I'll try this shortly.